lang/rust/rust-host.mk

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2023 Luca Barbato and Donald Hoskins
+
+# Rust Environmental Vars
+CONFIG_HOST_SUFFIX:=$(shell cut -d"-" -f4 <<<"$(GNU_HOST_NAME)")
+RUSTC_HOST_ARCH:=$(HOST_ARCH)-unknown-linux-$(CONFIG_HOST_SUFFIX)
+CARGO_HOME:=$(STAGING_DIR_HOST)/cargo
+
+# Support only a subset for now.
+RUST_ARCH_DEPENDS:=@(aarch64||mips||mipsel||mips64||mips64el||mipsel||powerpc64)
```

Comment:
If you can test on x86_64 and report back would be great. I do not want to try to debug x86 strange failures alone.

Reply:
I added x86-64 and built (host) rust succesfully, I also built maturin (again for host), and ripgrep(for target) succesfully. I have not run tested ripgrep on target yet (build and target arch are both x86-64), but atleast it compiled; so I think it's safe to add x86-64 to the list.

Reply:
Rebuilding with
`
make -j1 package/rust/host/{clean,compile} V=s
`
now

Reply:
Failure.

```
...
[267/3035] Linking CXX executable bin/llvm-tblgen
FAILED: bin/llvm-tblgen 
: && /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/bin/x86_64-openwrt-linux-musl-g++ -ffunction-sections -fdata-sections -fPIC -m64 -static -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -Wl,-rpath-link,/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/build/x86_64-unknown-linux-musl/llvm/build/./lib  -Wl,--gc-sections utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterInst.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Attributes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CallingConvEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenDAGPatterns.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenHwModes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenInstruction.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenMapTable.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenRegisters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenSchedule.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenTarget.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherOpt.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcher.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DecoderEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAPacketizerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DirectiveEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DXILEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/ExegesisEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/FastISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GICombinerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GlobalISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InfoByHwMode.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrDocsEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/IntrinsicEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptParserEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptRSTEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PredicateExpander.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PseudoLoweringEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CompressInstEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterBankEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SDNodeProperties.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SearchableTableEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetFeatureInfo.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/TableGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Types.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/VarLenCodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86DisassemblerTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86EVEX2VEXTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86FoldTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86MnemonicTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86ModRMFilters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86RecognizableInstr.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/WebAssemblyDisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CTagsEmitter.cpp.o -o bin/llvm-tblgen  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMSupport.a  lib/libLLVMTableGen.a  lib/libLLVMTableGenGlobalISel.a  lib/libLLVMTableGen.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  /usr/src/openwrt/staging_dir/host/lib/libz.a  lib/libLLVMDemangle.a && :
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_logic_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt19__throw_logic_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_length_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_length_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_out_of_rangePKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range_fmt(char const*, ...)':
functexcept.cc:(.text._ZSt24__throw_out_of_range_fmtPKcz+0x11d): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o): in function `std::_V2::error_category::_M_message[abi:cxx11](int) const':
system_error.cc:(.text._ZNKSt3_V214error_category10_M_messageB5cxx11Ei+0x72): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o):system_error.cc:(.text.unlikely._ZSt20__throw_system_errori+0xb5): more undefined references to `_Unwind_Resume' follow
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_catch.o): in function `__cxa_end_catch':
eh_catch.cc:(.text.__cxa_end_catch+0x54): undefined reference to `_Unwind_DeleteException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `base_of_encoded_value(unsigned char, _Unwind_Context*)':
eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x2e): undefined reference to `_Unwind_GetTextRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x37): undefined reference to `_Unwind_GetDataRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x40): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `parse_lsda_header(_Unwind_Context*, unsigned char const*, lsda_header_info*)':
eh_personality.cc:(.text._ZL17parse_lsda_headerP15_Unwind_ContextPKhP16lsda_header_info+0x1a): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `__gxx_personality_v0':
eh_personality.cc:(.text.__gxx_personality_v0+0x8c): undefined reference to `_Unwind_GetLanguageSpecificData'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0xd9): undefined reference to `_Unwind_GetIPInfo'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x39f): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3b2): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3c0): undefined reference to `_Unwind_SetIP'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_throw':
eh_throw.cc:(.text.__cxa_throw+0x38): undefined reference to `_Unwind_RaiseException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_rethrow':
eh_throw.cc:(.text.__cxa_rethrow+0x3d): undefined reference to `_Unwind_Resume_or_Rethrow'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(vterminate.o): in function `__gnu_cxx::__verbose_terminate_handler()':
vterminate.cc:(.text._ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x13b): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cow-stdexcept.o): in function `std::logic_error::logic_error(char const*)':
cow-stdexcept.cc:(.text._ZNSt11logic_errorC2EPKc+0x47): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cxx11-stdexcept.o): in function `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
cxx11-stdexcept.cc:(.text._ZNSt13runtime_errorC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x3c): undefined reference to `_Unwind_Resume'
collect2: error: ld returned 1 exit status
[274/3035] Building CXX object lib/MC/CMakeFiles/LLVMMC.dir/ELFObjectWriter.cpp.o
ninja: build stopped: subcommand failed.
thread 'main' panicked at '
command did not execute successfully, got: exit status: 1

build script failed, must exit now', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
	finished in 158.196 seconds
Build completed unsuccessfully in 0:08:18
make[2]: *** [Makefile:140: /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.built] Error 1
make[2]: Leaving directory '/usr/src/openwrt/feeds/packages/lang/rust'
time: package/feeds/packages/rust/host-compile#1436.55#181.43#526.01
    ERROR: package/feeds/packages/rust [host] failed to build.
make[1]: *** [package/Makefile:114: package/feeds/packages/rust/host/compile] Error 1
make[1]: Leaving directory '/usr/src/openwrt'
make: *** [/usr/src/openwrt/include/toplevel.mk:231: package/rust/host/compile] Error 2
[user@buildroot /usr/src/openwrt]$ 

```

Reply:
Let me know if there's something that I can do about this.

Reply:
> not sure if that warning is important..

It is not.

Reply:
> Failure.
> 
> ```
> ...
> [267/3035] Linking CXX executable bin/llvm-tblgen
> FAILED: bin/llvm-tblgen 
> : && /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/bin/x86_64-openwrt-linux-musl-g++ -ffunction-sections -fdata-sections -fPIC -m64 -static -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -Wl,-rpath-link,/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/build/x86_64-unknown-linux-musl/llvm/build/./lib  -Wl,--gc-sections utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterInst.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Attributes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CallingConvEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenDAGPatterns.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenHwModes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenInstruction.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenMapTable.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenRegisters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenSchedule.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenTarget.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherOpt.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcher.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DecoderEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAPacketizerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DirectiveEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DXILEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/ExegesisEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/FastISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GICombinerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GlobalISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InfoByHwMode.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrDocsEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/IntrinsicEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptParserEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptRSTEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PredicateExpander.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PseudoLoweringEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CompressInstEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterBankEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SDNodeProperties.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SearchableTableEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetFeatureInfo.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/TableGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Types.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/VarLenCodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86DisassemblerTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86EVEX2VEXTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86FoldTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86MnemonicTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86ModRMFilters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86RecognizableInstr.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/WebAssemblyDisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CTagsEmitter.cpp.o -o bin/llvm-tblgen  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMSupport.a  lib/libLLVMTableGen.a  lib/libLLVMTableGenGlobalISel.a  lib/libLLVMTableGen.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  /usr/src/openwrt/staging_dir/host/lib/libz.a  lib/libLLVMDemangle.a && :
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_logic_error(char const*)':
> functexcept.cc:(.text.unlikely._ZSt19__throw_logic_errorPKc+0x50): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_length_error(char const*)':
> functexcept.cc:(.text.unlikely._ZSt20__throw_length_errorPKc+0x50): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range(char const*)':
> functexcept.cc:(.text.unlikely._ZSt20__throw_out_of_rangePKc+0x50): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range_fmt(char const*, ...)':
> functexcept.cc:(.text._ZSt24__throw_out_of_range_fmtPKcz+0x11d): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o): in function `std::_V2::error_category::_M_message[abi:cxx11](int) const':
> system_error.cc:(.text._ZNKSt3_V214error_category10_M_messageB5cxx11Ei+0x72): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o):system_error.cc:(.text.unlikely._ZSt20__throw_system_errori+0xb5): more undefined references to `_Unwind_Resume' follow
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_catch.o): in function `__cxa_end_catch':
> eh_catch.cc:(.text.__cxa_end_catch+0x54): undefined reference to `_Unwind_DeleteException'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `base_of_encoded_value(unsigned char, _Unwind_Context*)':
> eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x2e): undefined reference to `_Unwind_GetTextRelBase'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x37): undefined reference to `_Unwind_GetDataRelBase'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x40): undefined reference to `_Unwind_GetRegionStart'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `parse_lsda_header(_Unwind_Context*, unsigned char const*, lsda_header_info*)':
> eh_personality.cc:(.text._ZL17parse_lsda_headerP15_Unwind_ContextPKhP16lsda_header_info+0x1a): undefined reference to `_Unwind_GetRegionStart'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `__gxx_personality_v0':
> eh_personality.cc:(.text.__gxx_personality_v0+0x8c): undefined reference to `_Unwind_GetLanguageSpecificData'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0xd9): undefined reference to `_Unwind_GetIPInfo'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x39f): undefined reference to `_Unwind_SetGR'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3b2): undefined reference to `_Unwind_SetGR'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3c0): undefined reference to `_Unwind_SetIP'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_throw':
> eh_throw.cc:(.text.__cxa_throw+0x38): undefined reference to `_Unwind_RaiseException'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_rethrow':
> eh_throw.cc:(.text.__cxa_rethrow+0x3d): undefined reference to `_Unwind_Resume_or_Rethrow'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(vterminate.o): in function `__gnu_cxx::__verbose_terminate_handler()':
> vterminate.cc:(.text._ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x13b): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cow-stdexcept.o): in function `std::logic_error::logic_error(char const*)':
> cow-stdexcept.cc:(.text._ZNSt11logic_errorC2EPKc+0x47): undefined reference to `_Unwind_Resume'
> /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cxx11-stdexcept.o): in function `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
> cxx11-stdexcept.cc:(.text._ZNSt13runtime_errorC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x3c): undefined reference to `_Unwind_Resume'
> collect2: error: ld returned 1 exit status
> [274/3035] Building CXX object lib/MC/CMakeFiles/LLVMMC.dir/ELFObjectWriter.cpp.o
> ninja: build stopped: subcommand failed.
> thread 'main' panicked at '
> command did not execute successfully, got: exit status: 1
> 
> build script failed, must exit now', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 	finished in 158.196 seconds
> Build completed unsuccessfully in 0:08:18
> make[2]: *** [Makefile:140: /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.built] Error 1
> make[2]: Leaving directory '/usr/src/openwrt/feeds/packages/lang/rust'
> time: package/feeds/packages/rust/host-compile#1436.55#181.43#526.01
>     ERROR: package/feeds/packages/rust [host] failed to build.
> make[1]: *** [package/Makefile:114: package/feeds/packages/rust/host/compile] Error 1
> make[1]: Leaving directory '/usr/src/openwrt'
> make: *** [/usr/src/openwrt/include/toplevel.mk:231: package/rust/host/compile] Error 2
> [user@buildroot /usr/src/openwrt]$ 
> ```

It looks like libunwind is not linked even if it is found, try to set the environment like `cmake.mk` does, _maybe_ it is really needed. IIRC openwrt builds llvm so either the rust toolchain can rely on it if it is modern enough or we can see how the env is set.

try to include cmake.mk before package.mk as it was done before for a quick test.

Reply:
I added include for cmake.mk before package.mk, re-building..

Reply:
that failed very fast:
```
Applying ./patches/0001-Update-xz2-and-use-it-static.patch using plaintext: 
patching file Cargo.lock
patching file src/bootstrap/Cargo.lock
patching file src/bootstrap/Cargo.toml
touch /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.prepared3146e04a64370ff32f007f8e9535f1f8_6664517399ebbbc92a37c5bb081b5c53
mkdir -p "/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/"
(cd /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/; CFLAGS="-O2 -I/usr/src/openwrt/staging_dir/host/include -I/usr/src/openwrt/staging_dir/hostpkg/include -I/usr/src/openwrt/staging_dir/target-x86_64_musl/host/include" CXXFLAGS="-O2 -I/usr/src/openwrt/staging_dir/host/include -I/usr/src/openwrt/staging_dir/hostpkg/include -I/usr/src/openwrt/staging_dir/target-x86_64_musl/host/include" LDFLAGS="-L/usr/src/openwrt/staging_dir/host/lib -L/usr/src/openwrt/staging_dir/hostpkg/lib -L/usr/src/openwrt/staging_dir/target-x86_64_musl/host/lib" cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER_LAUNCHER="" -DCMAKE_C_COMPILER="gcc" -DCMAKE_CXX_COMPILER_LAUNCHER="" -DCMAKE_CXX_COMPILER="g++" -DCMAKE_ASM_COMPILER_LAUNCHER="" -DCMAKE_ASM_COMPILER="gcc" -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/usr/src/openwrt/staging_dir/host/lib -L/usr/src/openwrt/staging_dir/hostpkg/lib -L/usr/src/openwrt/staging_dir/target-x86_64_musl/host/lib" -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/usr/src/openwrt/staging_dir/host/lib -L/usr/src/openwrt/staging_dir/hostpkg/lib -L/usr/src/openwrt/staging_dir/target-x86_64_musl/host/lib" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/usr/src/openwrt/staging_dir/host/lib -L/usr/src/openwrt/staging_dir/hostpkg/lib -L/usr/src/openwrt/staging_dir/target-x86_64_musl/host/lib" -DCMAKE_FIND_ROOT_PATH="/usr/src/openwrt/staging_dir/target-x86_64_musl/host;/usr/src/openwrt/staging_dir/hostpkg;/usr/src/openwrt/staging_dir/host" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr/src/openwrt/staging_dir/hostpkg -DCMAKE_PREFIX_PATH=/usr/src/openwrt/staging_dir/hostpkg -DCMAKE_SKIP_RPATH=TRUE -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_EXPORT_PACKAGE_REGISTRY=FALSE -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=TRUE -DCMAKE_FIND_USE_PACKAGE_REGISTRY=FALSE -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=TRUE -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=FALSE -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=TRUE -DCMAKE_GENERATOR="Ninja" /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/ )
CMake Warning:
  Ignoring extra path from command line:

   "/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/"


CMake Error: The source directory "/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make[2]: *** [Makefile:142: /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.configured] Error 1
make[2]: Leaving directory '/usr/src/openwrt/feeds/packages/lang/rust'
```

Reply:
not after, before.

Reply:
I moved it above host-build.mk as well-- results are up-coming shortly...

Reply:
llvm is build for the _host_ system not the target, so no matter the target it is built all the same way.

Reply:
I moved it above host-build.mk as well-- results are up-coming shortly... Build advances further than it did last time..

Reply:
the original author put cmake.mk on top and then included the rest to override back the defaults to the ones that use configure since the main build system has a configure script, then `x.py` build the initial build tools and then the rest, among it there is `llvm` and it is built for the host using cmake. But I tried with and w/out and at least on my systems it did not made any difference.

Reply:
No go there either. Stopped here:
```
[159/3035] Building C object lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_portable.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[266/3035] Linking CXX executable bin/llvm-tblgen
FAILED: bin/llvm-tblgen 
: && /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/bin/x86_64-openwrt-linux-musl-g++ -ffunction-sections -fdata-sections -fPIC -m64 -static -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -Wl,-rpath-link,/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/build/x86_64-unknown-linux-musl/llvm/build/./lib  -Wl,--gc-sections utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterInst.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Attributes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CallingConvEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenDAGPatterns.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenHwModes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenInstruction.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenMapTable.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenRegisters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenSchedule.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenTarget.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherOpt.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcher.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DecoderEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAPacketizerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DirectiveEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DXILEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/ExegesisEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/FastISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GICombinerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GlobalISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InfoByHwMode.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrDocsEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/IntrinsicEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptParserEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptRSTEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PredicateExpander.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PseudoLoweringEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CompressInstEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterBankEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SDNodeProperties.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SearchableTableEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetFeatureInfo.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/TableGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Types.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/VarLenCodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86DisassemblerTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86EVEX2VEXTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86FoldTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86MnemonicTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86ModRMFilters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86RecognizableInstr.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/WebAssemblyDisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CTagsEmitter.cpp.o -o bin/llvm-tblgen  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMSupport.a  lib/libLLVMTableGen.a  lib/libLLVMTableGenGlobalISel.a  lib/libLLVMTableGen.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  /usr/src/openwrt/staging_dir/host/lib/libz.a  lib/libLLVMDemangle.a && :
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_logic_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt19__throw_logic_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_length_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_length_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_out_of_rangePKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range_fmt(char const*, ...)':
functexcept.cc:(.text._ZSt24__throw_out_of_range_fmtPKcz+0x11d): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o): in function `std::_V2::error_category::_M_message[abi:cxx11](int) const':
system_error.cc:(.text._ZNKSt3_V214error_category10_M_messageB5cxx11Ei+0x72): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o):system_error.cc:(.text.unlikely._ZSt20__throw_system_errori+0xb5): more undefined references to `_Unwind_Resume' follow
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_catch.o): in function `__cxa_end_catch':
eh_catch.cc:(.text.__cxa_end_catch+0x54): undefined reference to `_Unwind_DeleteException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `base_of_encoded_value(unsigned char, _Unwind_Context*)':
eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x2e): undefined reference to `_Unwind_GetTextRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x37): undefined reference to `_Unwind_GetDataRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x40): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `parse_lsda_header(_Unwind_Context*, unsigned char const*, lsda_header_info*)':
eh_personality.cc:(.text._ZL17parse_lsda_headerP15_Unwind_ContextPKhP16lsda_header_info+0x1a): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `__gxx_personality_v0':
eh_personality.cc:(.text.__gxx_personality_v0+0x8c): undefined reference to `_Unwind_GetLanguageSpecificData'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0xd9): undefined reference to `_Unwind_GetIPInfo'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x39f): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3b2): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3c0): undefined reference to `_Unwind_SetIP'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_throw':
eh_throw.cc:(.text.__cxa_throw+0x38): undefined reference to `_Unwind_RaiseException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_rethrow':
eh_throw.cc:(.text.__cxa_rethrow+0x3d): undefined reference to `_Unwind_Resume_or_Rethrow'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(vterminate.o): in function `__gnu_cxx::__verbose_terminate_handler()':
vterminate.cc:(.text._ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x13b): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cow-stdexcept.o): in function `std::logic_error::logic_error(char const*)':
cow-stdexcept.cc:(.text._ZNSt11logic_errorC2EPKc+0x47): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cxx11-stdexcept.o): in function `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
cxx11-stdexcept.cc:(.text._ZNSt13runtime_errorC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x3c): undefined reference to `_Unwind_Resume'
collect2: error: ld returned 1 exit status
[273/3035] Building CXX object lib/MC/CMakeFiles/LLVMMC.dir/ELFObjectWriter.cpp.o
ninja: build stopped: subcommand failed.
thread 'main' panicked at '
command did not execute successfully, got: exit status: 1

build script failed, must exit now', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
	finished in 157.502 seconds
Build completed unsuccessfully in 0:07:41
make[2]: *** [Makefile:141: /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.built] Error 1
make[2]: Leaving directory '/usr/src/openwrt/feeds/packages/lang/rust'
time: package/feeds/packages/rust/host-compile#1438.56#181.58#489.08
    ERROR: package/feeds/packages/rust [host] failed to build.
make[1]: *** [package/Makefile:114: package/feeds/packages/rust/host/compile] Error 1
make[1]: Leaving directory '/usr/src/openwrt'
make: *** [/usr/src/openwrt/include/toplevel.mk:231: package/rust/host/compile] Error 2
[user@buildroot /usr/src/openwrt]$
```

Reply:
so libunwind striked again..

Reply:
try passing `--llvm-libunwind=in-tree` as configure option.

Reply:
```
include $(INCLUDE_DIR)/cmake.mk
include $(INCLUDE_DIR)/host-build.mk
include $(INCLUDE_DIR)/package.mk
```

Reply:
I added both, `--llvm-libunwind=in-tree` to configure opts, and cmake the way I had it earlier too when it failed. Thumbs up... Sorry, got back just a moment ago, had to attend to friend's wedding..

Reply:
No go still... Seems to have same issues as before even with in-tre libunwind..

```
running: "cmake" "--build" "." "--target" "install" "--config" "Release" "--" "-j" "8"
[3/3035] Building C object lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_dispatch.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[6/3035] Building C object lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[131/3035] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regerror.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[133/3035] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regfree.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[134/3035] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regstrlcpy.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[137/3035] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regcomp.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[139/3035] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regexec.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[156/3035] Building C object lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_neon.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[159/3035] Building C object lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_portable.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[265/3035] Linking CXX executable bin/llvm-tblgen
FAILED: bin/llvm-tblgen 
: && /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/bin/x86_64-openwrt-linux-musl-g++ -ffunction-sections -fdata-sections -fPIC -m64 -static -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -Wl,-rpath-link,/usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src/build/x86_64-unknown-linux-musl/llvm/build/./lib  -Wl,--gc-sections utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/AsmWriterInst.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Attributes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CallingConvEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenDAGPatterns.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenHwModes.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenInstruction.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenMapTable.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenRegisters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenSchedule.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CodeGenTarget.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcherOpt.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DAGISelMatcher.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DecoderEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DFAPacketizerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DirectiveEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/DXILEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/ExegesisEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/FastISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GICombinerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/GlobalISelEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InfoByHwMode.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/InstrDocsEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/IntrinsicEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptParserEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/OptRSTEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PredicateExpander.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/PseudoLoweringEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CompressInstEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterBankEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/RegisterInfoEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SDNodeProperties.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SearchableTableEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/SubtargetFeatureInfo.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/TableGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/Types.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/VarLenCodeEmitterGen.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86DisassemblerTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86EVEX2VEXTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86FoldTablesEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86MnemonicTables.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86ModRMFilters.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/X86RecognizableInstr.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/WebAssemblyDisassemblerEmitter.cpp.o utils/TableGen/CMakeFiles/llvm-tblgen.dir/CTagsEmitter.cpp.o -o bin/llvm-tblgen  -Wl,-rpath,"\$ORIGIN/../lib"  lib/libLLVMSupport.a  lib/libLLVMTableGen.a  lib/libLLVMTableGenGlobalISel.a  lib/libLLVMTableGen.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  /usr/src/openwrt/staging_dir/host/lib/libz.a  lib/libLLVMDemangle.a && :
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_logic_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt19__throw_logic_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_length_error(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_length_errorPKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range(char const*)':
functexcept.cc:(.text.unlikely._ZSt20__throw_out_of_rangePKc+0x50): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(functexcept.o): in function `std::__throw_out_of_range_fmt(char const*, ...)':
functexcept.cc:(.text._ZSt24__throw_out_of_range_fmtPKcz+0x11d): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o): in function `std::_V2::error_category::_M_message[abi:cxx11](int) const':
system_error.cc:(.text._ZNKSt3_V214error_category10_M_messageB5cxx11Ei+0x72): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(system_error.o):system_error.cc:(.text.unlikely._ZSt20__throw_system_errori+0xb5): more undefined references to `_Unwind_Resume' follow
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_catch.o): in function `__cxa_end_catch':
eh_catch.cc:(.text.__cxa_end_catch+0x54): undefined reference to `_Unwind_DeleteException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `base_of_encoded_value(unsigned char, _Unwind_Context*)':
eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x2e): undefined reference to `_Unwind_GetTextRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x37): undefined reference to `_Unwind_GetDataRelBase'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x40): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `parse_lsda_header(_Unwind_Context*, unsigned char const*, lsda_header_info*)':
eh_personality.cc:(.text._ZL17parse_lsda_headerP15_Unwind_ContextPKhP16lsda_header_info+0x1a): undefined reference to `_Unwind_GetRegionStart'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_personality.o): in function `__gxx_personality_v0':
eh_personality.cc:(.text.__gxx_personality_v0+0x8c): undefined reference to `_Unwind_GetLanguageSpecificData'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0xd9): undefined reference to `_Unwind_GetIPInfo'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x39f): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3b2): undefined reference to `_Unwind_SetGR'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: eh_personality.cc:(.text.__gxx_personality_v0+0x3c0): undefined reference to `_Unwind_SetIP'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_throw':
eh_throw.cc:(.text.__cxa_throw+0x38): undefined reference to `_Unwind_RaiseException'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(eh_throw.o): in function `__cxa_rethrow':
eh_throw.cc:(.text.__cxa_rethrow+0x3d): undefined reference to `_Unwind_Resume_or_Rethrow'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(vterminate.o): in function `__gnu_cxx::__verbose_terminate_handler()':
vterminate.cc:(.text._ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x13b): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cow-stdexcept.o): in function `std::logic_error::logic_error(char const*)':
cow-stdexcept.cc:(.text._ZNSt11logic_errorC2EPKc+0x47): undefined reference to `_Unwind_Resume'
/usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/bin/ld: /usr/src/openwrt/staging_dir/toolchain-x86_64_gcc-12.2.0_musl/lib/gcc/x86_64-openwrt-linux-musl/12.2.0/../../../../x86_64-openwrt-linux-musl/lib/libstdc++.a(cxx11-stdexcept.o): in function `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
cxx11-stdexcept.cc:(.text._ZNSt13runtime_errorC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x3c): undefined reference to `_Unwind_Resume'
collect2: error: ld returned 1 exit status
[272/3035] Building CXX object lib/MC/CMakeFiles/LLVMMC.dir/ELFObjectWriter.cpp.o
ninja: build stopped: subcommand failed.
thread 'main' panicked at '
command did not execute successfully, got: exit status: 1

build script failed, must exit now', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
	finished in 158.112 seconds
Build completed unsuccessfully in 0:07:28
make[2]: *** [Makefile:142: /usr/src/openwrt/build_dir/hostpkg/rustc-1.67.0-src//.built] Error 1
make[2]: Leaving directory '/usr/src/openwrt/feeds/packages/lang/rust'
time: package/feeds/packages/rust/host-compile#1435.29#185.19#476.41
    ERROR: package/feeds/packages/rust [host] failed to build.
make[1]: *** [package/Makefile:114: package/feeds/packages/rust/host/compile] Error 1
make[1]: Leaving directory '/usr/src/openwrt'
make: *** [/usr/src/openwrt/include/toplevel.mk:231: package/rust/host/compile] Error 2

```

Reply:
I'm doing another try- with default compiler settings this time. I had changed them a bit:
 - multi-threaded support
 - enable ccache
 - vendored rust crates

Maybe it helps to set these default again..

Reply:
Not helpful, same result.

Reply:
@oskarirauta 

can you add this to try? i can't test it because it doesn't have vps at the moment.
```
HOST_USE_NINJA:=1
CMAKE_INSTALL:=1
```

Reply:
to rust/Makefile, below PKG_HOST_ONLY:=1 ?

Reply:
Same result.

Reply:
@oskarirauta 
thank you for trying it. right now I can't say anything without making it. it's possible that too many changes in rust have been made

Reply:
https://github.com/rust-lang/rust/issues/79609

Helpful..?

Reply:
What is your host system? Here building llvm works just fine.

Reply:
alpine linux

Reply:
install libunwind and make sure libunwind.a is in the host.

Reply:
I already have it- like I also do have rust and cargo on the host system. I also updated my rust package to rust 1.67.1 hoping that would be helpful.

```
[user@buildroot /usr/src/openwrt]$ ls /usr/lib/libunwind**
/usr/lib/libunwind-coredump.a         /usr/lib/libunwind-ptrace.a         /usr/lib/libunwind-setjmp.so.0      /usr/lib/libunwind.a
/usr/lib/libunwind-coredump.so        /usr/lib/libunwind-ptrace.so        /usr/lib/libunwind-setjmp.so.0.0.0  /usr/lib/libunwind.so
/usr/lib/libunwind-coredump.so.0      /usr/lib/libunwind-ptrace.so.0      /usr/lib/libunwind-x86_64.a         /usr/lib/libunwind.so.1
/usr/lib/libunwind-coredump.so.0.0.0  /usr/lib/libunwind-ptrace.so.0.0.0  /usr/lib/libunwind-x86_64.so        /usr/lib/libunwind.so.1.0
/usr/lib/libunwind-generic.a          /usr/lib/libunwind-setjmp.a         /usr/lib/libunwind-x86_64.so.8      /usr/lib/libunwind.so.8
/usr/lib/libunwind-generic.so         /usr/lib/libunwind-setjmp.so        /usr/lib/libunwind-x86_64.so.8.0.1  /usr/lib/libunwind.so.8.0.1
[user@buildroot /usr/src/openwrt]$ 

```

Reply:
I tried building on a alpine/edge and even the host perl failed to build...

Reply:
@lu-zero yes, mine though builds, but I have modified perl build process.

```
95a96,109
> define Build/Prepare
> 	$(Build/Prepare/Default)
> 	mkdir -p $(PKG_BUILD_DIR)/perl5lib/perl5 $(PKG_BUILD_DIR)/cpan/lib $(PKG_BUILD_DIR)/ext/lib
> 	cp -R ./alpine/perl5/** $(PKG_BUILD_DIR)/perl5lib/perl5/
> 	cp ./alpine/lib.pm $(PKG_BUILD_DIR)/cpan/lib/
> 	ln -s $(PKG_BUILD_DIR)/perl5lib/perl5 $(PKG_BUILD_DIR)/cpan/lib/perl5
> 	ln -s $(PKG_BUILD_DIR)/perl5lib/perl5 $(PKG_BUILD_DIR)/dist/lib/perl5
> 	ln -s $(PKG_BUILD_DIR)/perl5lib/perl5 $(PKG_BUILD_DIR)/ext/lib/perl5
> 	ln -s $(PKG_BUILD_DIR)/perl5lib $(PKG_BUILD_DIR)/lib
> 	[ -L "$(PKG_BUILD_DIR)/../lib" ] || {
> 		ln -s $(PKG_BUILD_DIR)/perl5lib $(PKG_BUILD_DIR)/../lib
> 	}
> endef
```

That's patch against perl's makefile on package tree- not against perl's makefile on actual perl source.

And, ./alpine/perl5 is a directory containing perl5 files required for building, collection of various files that I gathered here and there that got the job done- a bit big collection, but as it's mostly for personal use.....

[files.tar.gz](https://github.com/openwrt/packages/files/10777377/files.tar.gz)

Another package that needs some help to compile, is git.. Or atleast did at some point, git got recently updated and not sure if I had still my patches there.. But very likely you won't need that for testing this..

Reply:
I do not know reason why this happened for perl on alpine linux. Some other stuff that then again we're impossible to build on archlinux builds without issues on alpine.

Reply:
Origin of these errors is very strange- g++ should automaticly link libstdc++ but I just took part to another project (building openwrt on openwrt) - and had same problems while building toolchain, issues were fixed by manually adding -lstdc++ to LDFLAGS **even when it should be there automaticly by g++** - once I have built _all_ packages.. I'll take a attempt to add rust-lang to my buildroot, most likely I will be facing same issues, but now I could try adding -lstdc++ to ldflags to see if it helps..

I don't know if this problem is musl related or what, but hopefully if someone else finds himself in situation like this, he can find this little tip (though, I haven't yet tried it, but I have high hopes up)

Reply:
That unfortunately did not help at all :(

Reply:
Okay- I was almost giving up - and after some changes to Makefile, alongside with few patches imported from alpine's rust build, I forgot I had it building on the background and was looking for something else in web, when I thought that I'd remove rust-lang from my tree and got back to terminal- and what do you know.. There it is.. Compiling- past the point where it previously failed..

So, I stopped compilation, removed patches, reverted **some** of my setup in Makefile to as it was, and started a new build- let's see if it compiles now, if not- I'll just add piece by piece, until I find out what the secret recipe is...

This though, when being progress- cannot be said a success yet; whilst getting past _that_ point, which was about 264th part of that longish 3035 parts compilation; does not mean that it would reach to the end.. or that there wouldn't be other problems.. But getting there- slowly :)

Reply:
That was fast... Got past that point again and it keeps compiling.. Well.. Here are my changes to Makefile..

```
# Target Flags
TARGET_CONFIGURE_ARGS = \
  --set=target.$(RUSTC_TARGET_ARCH).ar=$(TARGET_AR) \
  --set=target.$(RUSTC_TARGET_ARCH).cc=$(TARGET_CC_NOCACHE) \
  --set=target.$(RUSTC_TARGET_ARCH).cxx=$(TARGET_CXX_NOCACHE) \
  --set=target.$(RUSTC_TARGET_ARCH).linker=$(TARGET_CC_NOCACHE) \
  --set=target.$(RUSTC_TARGET_ARCH).musl-root=$(TOOLCHAIN_DIR) \
  --set=target.$(RUSTC_TARGET_ARCH).ranlib=$(TARGET_RANLIB) \
  --set=target.$(RUSTC_TARGET_ARCH).crt-static=false \
  --set=rust.codegen-units=1 \
  --set=rust.codegen-units-std=1 \
  --set=rust.description=openwrt

# CARGO_HOME is an environmental
HOST_CONFIGURE_OPTS += CARGO_HOME="$(CARGO_HOME)" PKG_CONFIG_ALLOW_CROSS=1
HOST_MAKE_VARS += PKG_CONFIG_ALLOW_CROSS=1
```

summary:
crt-static=false
rust.codegen-units=1, rust.codegen-units-std
And then I added PKG_CONFIG_ALLOW_CROSS=1 to env..
and even gave name to distribution..
I removed those patches for now...

And with that, I have currently passed way over 400th part of build process.. Remarkable.

I copied these from Alpine's and gentoo's recipe (and one proposal of gentoo recipe changes), and absolutely have no idea how they effect on build, will these changes render recipe unusable on glibc hosts, or are there downsides of building it like this- I was thinking, that if I get through this and can have a working rust toolchain.. This could assist also cross-compiling toolchain for target as well- and in case that these have effect on glibc that prevents building, or makes it anyway worse on glibc builds- maybe there could be a setting in conf, to enable these easily, something smarter than "_Build on musl host_" but similar anyway.... But how optimistic I am here, build process is still on-going and will be on-going for some time now, so I'll leave it to build on a tmux'd shell and check the progress later. I maybe should sit here and stare at it, as usually things break immediately after you leave the chair... but it's taken all night already, local time is 5.25 am, so I'll hit the bed with high hopes of success. Wish me luck...

Reply:
Who am I fooling? Like I could sleep when it's this thrilling. 1 hour 55 minutes to build it, success- now it just started to build ripgrep - exitement continues, it remains to be seen, if it is able to compile something also..

But why on earth, everything on rust needs so much space? rust compiled binaries need huge spaces, starting from megabytes, even for small things. And rust build source takes 8.2gb :O

Reply:
Okay, not that good then. ripgrep failed.
can't find crate core, std......

Reply:
Hey @oskarirauta , could you please take a deep breath and let us know how you ended up once you prepared the stuff and compiled it? We don't need to know immediately that you are compiling it. :) Things takes time, so it would be good if you wait with your respond, too once you are sure that you want to comment it.

Right now, you are letting us know about every single thing that you are doing, and thats not necessary since everyone involved in this PR gets a notification (which could be forwarded to e-mail) once you comment it. Right now, I got 5 email notifications (one per 2 hours) from you. 

Yes, I could unsubscribe from this PR, but that's not also wanted.

Reply:
libcore cannot be found anywhere from staging dir...

Reply:
Please stop squatting this PR.